### PR TITLE
replacing use of deprecated apache http client calls

### LINKED ISSUE #68: Deprecated Elements
In Servlet Sample uses Apache's http.core. However Apache elements deprecated for Over 25 APIs. Do you think to migrate ssl.HttpURL(s)Connection package

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #68 

Replacing use of Apache's deprecated `DefaultHttpClient` with `HttpClientBuilder`

### CODE REVIEW

**Comment by patflynn** [comment]
LGTM

**Comment by loosebazooka** [comment]
I think the problem is deeper than this :

Deprecation notice : 
http://developer.android.com/about/versions/marshmallow/android-6.0-changes.html#behavior-apache-http-client

A possible workaround :
http://stackoverflow.com/questions/29294479/android-deprecated-apache-module-httpclient-httpresponse-etc

**Comment by etanshaul** [comment]
I see. I didn't catch that support for the apache http client was removed altogether. I'll update this.

**Comment by etanshaul** [comment]
updated sample code removing all uses of the apache http client

File: HelloWorld/README.md [hunk 1]
```diff
@@ -54,28 +54,66 @@ class ServletPostAsyncTask extends AsyncTask<Pair<Context, String>, Void, String
...
-            nameValuePairs.add(new BasicNameValuePair("name", name));
-            httpPost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
-
-            // Execute HTTP Post Request
-            HttpResponse response = httpClient.execute(httpPost);
-            if (response.getStatusLine().getStatusCode() == 200) {
-                return EntityUtils.toString(response.getEntity());
+            // Set up the request
+            URL url = new URL("http://10.0.2.2:8080/hello");
+            HttpURLConnection connection = (HttpURLConnection)url.openConnection();
```
Comment (elharo) [inline HelloWorld/README.md]: space after cast

**Comment by elharo** [comment]
lgtm

**Comment by loosebazooka** [comment]
lgtm

**Pull request merged**